src/Cogworks.AzureSearch/Operations/IndexOperation.cs

```diff
@@ -29,7 +29,8 @@ public async Task<bool> IndexExistsAsync()
             }
             catch (Exception exception)
             {
-                throw new IndexExistsException(exception.Message, exception.InnerException);
+                var domainException = new IndexExistsException(exception.Message, exception.InnerException);
```

Comment:
What's the purpose of this split up?

Reply:
this was also my question, I will link you to the article: https://cezarypiatek.github.io/post/exceptions-usages-analyzer/

in general I wanted to disable it in editorconfig but it looks like this analyzer does not support disabling rules. I will reply if I find something related with this split.

Reply:
OK thats the case: https://cezarypiatek.github.io/post/exceptions-usages-analyzer/#context-will-save-you

and this rule make it failure: 
![image](https://user-images.githubusercontent.com/29370044/118117534-0c015600-b3ec-11eb-903a-4442cc11bc3d.png)

Reply:
we cannot do to much - not including package downgrade -> but we don't want to do it.